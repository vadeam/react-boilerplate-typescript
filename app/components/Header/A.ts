import NormalA from 'components/A'
import styled from 'styles/styled-components'

const A = styled(NormalA)`
  padding: 2em 0;
`

export default A
