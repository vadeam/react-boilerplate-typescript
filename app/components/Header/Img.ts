import NormalImg from 'components/Img'
import styled from 'styles/styled-components'

const Img = styled(NormalImg)`
  width: 100%;
  margin: 0 auto;
  display: block;
`

export default Img
