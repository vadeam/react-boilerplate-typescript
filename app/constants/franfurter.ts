export const DEFAULT_BASE_CURRENCY = 'EUR'
