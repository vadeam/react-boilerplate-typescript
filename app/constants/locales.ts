export const LOCALE_DE = 'de'
export const LOCALE_EN = 'en'
export const DEFAULT_LOCALE = LOCALE_EN

export const appLocales: Locales[] = [LOCALE_EN, LOCALE_DE]
export const CUSTOM_LOCALE_STORAGE_KEY = 'custom-locale'
