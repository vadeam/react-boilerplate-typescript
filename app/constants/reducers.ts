export const REDUCER_KEY_APP = 'app'
export const REDUCER_KEY_LANGUAGE = 'language'
export const REDUCER_KEY_ROUTER = 'router'
export const REDUCER_KEY_HOME = 'home'
// for testing purposes
export const REDUCER_KEY_TEST = 'test'
