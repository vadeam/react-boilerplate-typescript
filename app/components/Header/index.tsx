import * as React from 'react'
import { FormattedMessage } from 'react-intl'

import A from './A'
import Banner from './banner.jpg'
import HeaderLink from './HeaderLink'
import Img from './Img'
import messages from './messages'
import NavBar from './NavBar'

function Header() {
  return (
    <div>
      <A href="https://www.reactboilerplate.com/">
        <Img src={Banner} alt="react-boilerplate - Logo" />
      </A>
      <NavBar>
        <HeaderLink to="/">
          <FormattedMessage {...messages.home} />
        </HeaderLink>
        <HeaderLink to="/features">
          <FormattedMessage {...messages.features} />
        </HeaderLink>
      </NavBar>
    </div>
  )
}

export default Header
