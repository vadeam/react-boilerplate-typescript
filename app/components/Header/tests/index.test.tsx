import { render } from '@testing-library/react'
import { ConnectedRouter } from 'connected-react-router'
import { createMemoryHistory } from 'history'
import React from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'

import configureStore from '../../../configureStore'
import Header from '../index'
import { DEFAULT_LOCALE } from '../../../constants/locales'

describe('<Header />', () => {
  const history = createMemoryHistory()
  const store = configureStore({}, history)

  it('should render a div', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <ConnectedRouter history={history}>
            <Header />
          </ConnectedRouter>
        </IntlProvider>
      </Provider>,
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
