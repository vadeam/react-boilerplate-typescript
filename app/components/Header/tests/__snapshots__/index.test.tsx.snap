// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render a div 1`] = `
<div>
  <a
    class="sc-bdnxRM sc-gtsrHT eBHLDh jgFwQc"
    href="https://www.reactboilerplate.com/"
  >
    <img
      alt="react-boilerplate - Logo"
      class="sc-dlnjwi fZWUhH"
      src="IMAGE_MOCK"
    />
  </a>
  <div
    class="sc-hKFxyN dIIkaJ"
  >
    <a
      class="sc-eCApnc hRmJFz"
      href="/"
    >
      <span>
        Home
      </span>
    </a>
    <a
      class="sc-eCApnc hRmJFz"
      href="/features"
    >
      <span>
        Features
      </span>
    </a>
  </div>
</div>
`;
